### PR TITLE
The message, user, and conversation classes have been abstracted into a single 'events' class.

### PR DESCRIPTION
**Pull request by VillanuevaK**
-Created an event class that takes the data from the message, user, and conversation classes and holds all the data in the class. This was done so that the chronological sorting of the objects would be easier and so that they are further abstracted and easier to use all at once.

-Created an event store class that converts the user, message, and conversation classes into event classes. Will eventually use it to permanently store the event objects.

-In the activity servlet I made a the list of events accessible to the activity.jsp file. I removed unnecessary code and imports now that we are only dealing with event objects. 

-In the activity.jsp file I removed unnecessary code and imports now that we are only dealing with event objects. What appears on the website is now almost exactly the same as the example, except for the links to the conversation, bold font for the time, and the formatting of the time.

-In the UserStore file I added a function so that I would have access to all the users.

-In the MessageStore file I added a function that allows me to access all the messages that currently exist.

-In the ConversationStore file I added a function that allows me to access a conversation given a conversation ID.

-In the ServerStartupListener.java file I made it so that the EventStore instance can access the list of users, conversations, and messages in memory.

### CODE REVIEW

File: src/main/java/codeu/controller/ActivityServlet.java [hunk 1]
```diff
@@ -70,15 +71,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
 	   /* Get all conversations. */
 	   List<Conversation> conversations = conversationStore.getAllConversations();
 
-	   /* Make the conversations variable accesible to the jsp file. */           
+	   /*Get all users. */
+	   List <User> users = userStore.getAllUsers();
```
Comment (yufangqq) [inline src/main/java/codeu/controller/ActivityServlet.java]: nit:
Note the spacing

/* Get all users. */
List<User> users = userStore.getAllUsers();

File: src/main/webapp/WEB-INF/view/activity.jsp [hunk 2]
```diff
@@ -61,29 +63,41 @@ UserStore userStore = (UserStore) request.getAttribute("userStore");
   <ul>
     <%
        /* Used to format the time since the getCreationTime() method returns an instance, not a string. */
-       DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime( FormatStyle.SHORT )
-         .withLocale( Locale.US )
-         .withZone( ZoneId.systemDefault() );
+      DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime( FormatStyle.SHORT )
+        .withLocale( Locale.US )
+        .withZone( ZoneId.systemDefault() );
```
Comment (yufangqq) [inline src/main/webapp/WEB-INF/view/activity.jsp]: We normally do NOT put extra spaces within the paranthesis, i.e., withZoen(ZoneId.systemDefault()).

File: src/main/webapp/WEB-INF/view/activity.jsp [hunk 3]
```diff
@@ -61,29 +63,41 @@ UserStore userStore = (UserStore) request.getAttribute("userStore");
...
-           <li><%= formatttedCreationTime + ": " + userStore.getUser(message.getAuthorId()).getName() +
-            " sent a message in " + title + ": " + "\"" + message.getContent() + "\"" %></li>  
+      <li><%= formatttedConversationCreationTime + ": " + userStore.getUser(conversation.getOwnerId()).getName() 
+        + " created a new conversation: " + conversation.getTitle() %></li>  
+      <%
+        /* Get the messages of the conversations. */
+        List<Message> messages = messageStore.getMessagesInConversation(conversation.getId());
+        for (Message message : messages) {
+      %>
+          <%
```
Comment (yufangqq) [inline src/main/webapp/WEB-INF/view/activity.jsp]: %>
<%

Here are unnecessary

**Review by yufangqq — APPROVED** [review]
Did you run https://github.com/google/google-java-format against the Java files?

File: src/main/java/codeu/controller/ServerStartupListener.java [hunk 4]
```diff
@@ -31,6 +33,9 @@ public void contextInitialized(ServletContextEvent sce) {
       List<Message> messages = PersistentStorageAgent.getInstance().loadMessages();
       MessageStore.getInstance().setMessages(messages);
 
+      /* This lets the EventStore access the user, conversation, and message lists. */
+      EventStore.getInstance().setEvents(users, conversations, messages);
```
Comment (yufangqq) [inline src/main/java/codeu/controller/ServerStartupListener.java:37]: Note, I think one limitation with this approach of EventStore encapsulating all the users/conversations/messages is that if a NEW user/conversation/message is created, the EventStore becomes outdated, it does NOT automatically get that NEW user/conversation/message.

And if you want to keep EventStore up-to-date, every time NEW user/conversation/message is created, you not only have to update UserStore/ConverstaionStore/MessageStore, you have to update the EventStore.

And if you want to persist all the events in the database, for example, every time a new User is created, that user is inserted into both 'user' table as well as 'event' table. So duplicated data not desirable.

File: src/main/java/codeu/model/data/Event.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+import codeu.model.store.basic.UserStore;
+
+/** 
+ *  This class abstracts any object so it can be a User, Conversation, or Message. 
+ *  Made so a list can  easily hold and sort all types of objects at once.
+ */
+public class Event{
+
+	/* Denotes the type of event this is. 'c' for conversation, 'm' for message, and 'u' for user. */
+	private final Character eventType;
```
Comment (yufangqq) [inline src/main/java/codeu/model/data/Event.java]: Normally we use a ENUM for such thing like eventType.

File: src/main/java/codeu/model/data/Event.java [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+	  	return dateTime_ToString(userCreationTime) + " PST: " + nameOfUser + " joined!";
+	  }
+	  else{
+	  	System.err.println("This object has no event type. This is impossible. I don't know how you even caused this error");
+	  	return null;
+	  }
+	}
+
+    /** Formats time of type Instant into a string. */
+	public String dateTime_ToString(Instant unformattedTime) {
```
Comment (yufangqq) [inline src/main/java/codeu/model/data/Event.java]: Either
dateTimeToString()
or
simply toString(), it  takes an Instant as parameter, so it is pretty obvious it is converting a time to string.

File: src/main/java/codeu/model/data/Event.java [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    private final ConversationStore conversationStore;
+  
+  /**
+   * Constructs a new event using a Conversation object.
+   *
+   * @param conversation an object of type Conversation
+   */
+	public Event(Conversation conversation) {
+      userStore = UserStore.getInstance();
+      conversationStore = ConversationStore.getInstance();
```
Comment (yufangqq) [inline src/main/java/codeu/model/data/Event.java]: Again, I understand your approach of using the notion of Event for your sorting purpose.
You can proceed with this change.
But I want to talk to you at some point about the design drawback of this approach. How code and data need to be duplicated to support this, which is undesirable.

File: src/main/java/codeu/model/store/basic/EventStore.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    conversations = new ArrayList<>();
+    messages = new ArrayList<>();
+    users = new ArrayList<>();
+    events = new ArrayList<>();
+  }
+
+  /** Access the current set of events known to the application in chronological order. */
+  public List<Event> getAllEventsSorted() {
+  	/* Sort events based on the time that they were created. */
+  	events.sort(Comparator.comparingLong(event -> event.getCreationTime()));
```
Comment (yufangqq) [inline src/main/java/codeu/model/store/basic/EventStore.java:78]: NICE! use of the Comparator.

**Review by yufangqq — COMMENTED** [review]
Great work! You have demonstrated good understanding of the servelet code!

**Review by yufangqq — APPROVED** [review]

File: src/main/java/codeu/controller/ServerStartupListener.java (hunk 4 above)
Comment (VillanuevaK) [reply src/main/java/codeu/controller/ServerStartupListener.java:37]: I was thinking about these things and I think updating the EventStore will be the best option. As far as duplicate data, I'm going to think about how to circumvent that, but in my opinion, the ease of dealing with just the event class and how much more clean it makes the code in the activity.jsp file outweighs the cons.

File: src/main/java/codeu/model/data/Event.java (hunk 5 above)
Comment (VillanuevaK) [reply src/main/java/codeu/model/data/Event.java]: Will change this.

File: src/main/java/codeu/model/data/Event.java (hunk 6 above)
Comment (VillanuevaK) [reply src/main/java/codeu/model/data/Event.java]: Got it. Forgot about overloading for some reason.

**Pull request merged**